src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,348 @@
+        public static readonly FixIdDefinition UseCollectionInitializerDiagnosticId;
+
+        [Export]
+        [FixId(RemoveUnusedImportsFixId)]
+        [Name(RemoveUnusedImportsFixId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [ExportMetadata("EnableByDefault", true)]
+        [LocalizedName(typeof(CSharpVSResources), nameof(CSharpVSResources.Remove_unnecessary_usings))]
+        public static readonly FixIdDefinition RemoveUnusedImports;
+
+        [Export]
+        [FixId(SortImportsFixId)]
+        [Name(SortImportsFixId)]
+        [Order(After = RemoveUnusedImportsFixId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [ExportMetadata("EnableByDefault", true)]
+        [LocalizedName(typeof(CSharpVSResources), nameof(CSharpVSResources.Sort_usings))]
+        public static readonly FixIdDefinition SortImports;
+
+        public override Task<bool> FixAsync(ICodeCleanUpScope scope, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        {
+            switch (scope)
+            {
+                case TextBufferCodeCleanUpScope textBufferScope:
+                    return FixTextBufferAsync(textBufferScope, context, cancellationToken);
+                case IVsHierarchyCodeCleanupScope hierarchyContentScope:
+                    return FixHierarchyContentAsync(hierarchyContentScope, context, cancellationToken);
+                default:
+                    return Task.FromResult(false);
+            }
+        }
+
+        private Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        {
+            // TODO: this one will be implemented later
+            // https://github.com/dotnet/roslyn/issues/30165
+            var hierarchy = hierarchyContent.Hierarchy;
+            if (hierarchy == null)
+            {
+                // solution
+                return Task.FromResult(false);
+            }
+
+            var itemId = hierarchyContent.ItemId;
+
+            if (hierarchy.GetCanonicalName(itemId, out var path) == 0)
+            {
+                var attr = File.GetAttributes(path);
+                if (attr.HasFlag(FileAttributes.Directory))
+                {
+                    // directory
+                }
+                else
+                {
+                    // document
+                }
+            }
+
+            return Task.FromResult(false);
+        }
+
+        private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBufferScope, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        {
+            using (var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(context.OperationContext.UserCancellationToken, cancellationToken))
```

Comment:
@olegtk this API seems a bit weird. is context.OperationContext.UserCancellationToken and given cancellationToken different thing? why is it giving us 2 cancellation token?

Reply:
yep, we need to drop one. @crmann1 ?